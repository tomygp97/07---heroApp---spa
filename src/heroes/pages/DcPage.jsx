import { HeroList } from "../components"

export const DcPage = () => {

  const publisher = 'DC Comics'

  return (

      <>
        <h1>DC Comics</h1>
        <hr />

        <HeroList publisher={ publisher }/>
      
      </>

    )
}
