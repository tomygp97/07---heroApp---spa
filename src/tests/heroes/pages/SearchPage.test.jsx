import { render, screen, fireEvent } from "@testing-library/react"
import { MemoryRouter, useNavigate } from "react-router-dom"
import { SearchPage } from "../../../heroes/pages/SearchPage"


const mockUseNavigate = jest.fn();

jest.mock( "react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockUseNavigate
}) );

describe('Tests in <SearchPage />', () => {

    beforeEach( () => jest.clearAllMocks() );

    test('should shown correctly with default values', () => {


        const { container } = render( 
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
         );

        expect( container ).toMatchSnapshot();
        //  screen.debug();


    });


    test('should show Batman and input with queryString value', () => {


        const { container } = render( 
            <MemoryRouter initialEntries={["/search?q=batman"]}>
                <SearchPage />
            </MemoryRouter>
         );

        const input = screen.getByRole("textbox");
        expect( input.value ).toBe("batman");

        const img = screen.getByRole("img");
        expect( img.src ).toBe( "http://localhost/assets/heroes/dc-batman.jpg" )

        //  screen.debug();
    });


    test('should keep display none after search a hero', () => {


        const { container } = render( 
            <MemoryRouter initialEntries={["/search?q=batman"]}>
                <SearchPage />
            </MemoryRouter>
         );

        const dispNone = screen.getByLabelText("hidden");
        // console.log(dispNone.style._values.display);
        // expect( dispNone.style._values.display ).toBe( "none" );
        expect( dispNone.style.display ).toBe( "none" );
        // screen.debug();

    });


    test('should show an error if hero is not finded (batman123)', () => {

        const { container } = render( 
            <MemoryRouter initialEntries={["/search?q=batman123"]}>
                <SearchPage />
            </MemoryRouter>
         );

        const dispNone = screen.getByLabelText("hidden");
        expect( dispNone.style.display ).toBe( "" );
    });


    test('should call navigate to the new screen', () => {

        const { container } = render( 
            <MemoryRouter initialEntries={["/search"]}>
                <SearchPage />
            </MemoryRouter>
         );

        const input = screen.getByRole("textbox");
        const form  = screen.getByRole("form");
         

        // fireEvent.input( input, { target: {value: "superman"} } );
        fireEvent.change( input, { target: {name: "searchText", value: "superman"} } )
        fireEvent.submit( form );

        expect( mockUseNavigate ).toHaveBeenCalledWith( "?q=superman" );

        // screen.debug();
    });


});