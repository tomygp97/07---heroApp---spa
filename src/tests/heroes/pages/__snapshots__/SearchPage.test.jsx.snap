// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <SearchPage /> should shown correctly with default values 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a Hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeInRight"
      >
        Search a Hero
      </div>
      <div
        aria-label="hidden"
        class="alert alert-danger animate__animated animate__fadeInRight"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
